lighthouse-viewer/app/src/main.js

```diff
@@ -0,0 +1,504 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* global window, document */
+
+const ReportGenerator = require('../../../lighthouse-core/report/report-generator');
+// TODO: We only need getFilenamePrefix from asset-saver. Tree shake!
+const getFilenamePrefix = require('../../../lighthouse-core/lib/asset-saver.js').getFilenamePrefix;
+
+const firebase = require('firebase/app');
+require('firebase/auth');
+const idb = require('idb-keyval');
+
+// Polyfills
+// TODO: dynamically load this based on feature detection.
+require('whatwg-fetch');
+const URLSearchParams = require('url-search-params');
+
+const LH_CURRENT_VERSION = require('../../../package.json').version;
+const APP_URL = `${location.origin}${location.pathname}`;
+
+class Logger {
+  constructor(selector) {
+    this.el = document.querySelector(selector);
+  }
+
+  /**
+   * Shows a butter bar.
+   * @param {!string} msg The message to show.
+   * @param {boolean=} autoHide True to hide the message after a duration.
+   *     Default is true.
+   */
+  log(msg, autoHide = true) {
```

Comment:
is this default param transpiled out by the closure step or are we only supporting browsers with support for them?